src/Bicep.Core/Semantics/ObjectParser.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Intermediate;
+using Bicep.Core.Parsing;
+using Bicep.Core.TypeSystem;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using SharpYaml.Serialization;
+
+namespace Bicep.Core.Semantics
+{
+    public abstract class ObjectParser : IObjectParser
+    {
+        public bool TryExtractFromObject(string fileContent, string? tokenSelectorPath, IPositionable[] positionable, [NotNullWhen(false)] out ErrorDiagnostic? errorDiagnostic, out JToken newToken)
+        {
+            errorDiagnostic = null;
+            newToken = this.ExtractTokenFromObject(fileContent);
+            if (newToken is not { })
+            {
+                // Instead of catching and returning the YML parse exception, we simply return a generic error.
+                // This avoids having to deal with localization, and avoids possible confusion regarding line endings in the message.
+                errorDiagnostic = this.GetExtractTokenErrorType(positionable[0]);
+                return false;
+            }
+            if (tokenSelectorPath is not null)
+            {
+                return this.TryExtractFromTokenByPath(newToken, tokenSelectorPath, positionable[1], out errorDiagnostic, out newToken);
+            }
+            return true;
+        }
+        public abstract JToken ExtractTokenFromObject(string fileContent);
+
+        public abstract ErrorDiagnostic GetExtractTokenErrorType(IPositionable positionable);
+
+        public bool TryExtractFromTokenByPath(JToken token, string tokenSelectorPath, IPositionable positionable, [NotNullWhen(false)] out ErrorDiagnostic? errorDiagnostic, out JToken newToken)
```

Comment:
`private` on this unless you need it to be public

Reply:
that causes....
<img width="444" alt="image" src="https://user-images.githubusercontent.com/9027725/232820810-1572b3ac-a60d-439d-86a6-25a7ee7d6ef0.png">

Reply:
fixed